libs/ml/include/ml/core/graph.hpp

```diff
@@ -74,6 +75,7 @@ class Graph
   NodePtrType GetNode(std::string const &node_name) const;
   void        SetInput(std::string const &node_name, TensorType data);
   void        ResetGraphCache(std::shared_ptr<Node<T>> const &n, bool input_size_changed);
+  void        ResetWholeGraphCache(bool input_size_changed);
```

Comment:
just name this method ResetGraphCache - makes sense to have it as an overloaded function